### DISCUSSION TITLE
Intercept HTTP requests in standalone browser script

### DISCUSSION THREAD

**Discussion by sorasful** [mswjs/msw#2192] [Q&A] [answered]
Hello,

I have browsed through various repositories and discussions but couldn't find an answer to my question.
I wanted to know if it is possible with MSW to create a JavaScript script that I serve to clients, which they load using a <script> tag like this:

```html
<script src="https://mydomain/script.js" defer=""></script>
```
and make it so that whenever a request is intercepted, a console log is displayed with the URL to which the request was made.

I'm not sure if this is entirely off-topic or not.

Thank you for your time and assistance.

**Comment by kettanaito** [comment]
Hi, @sorasful. 

Should the `/script.js` _initiate_ the interception? Or should the `/script.js` itself be intercepted on the page?

**Reply by sorasful** [reply]
Hey @kettanaito  ! To me the `script.js` should initiate the interception.

**Reply by kettanaito** [reply, marked as answer]
Got it!

That's not going to work. MSW uses the Service Worker API in the browser. Since it allows you to intercept the traffic on a page, it has to be secure. One of its security measures is that the consumer has to serve the worker script. Just including a <script> tag will not do anything.

I generally advise against this pattern. Cross-domain network control is dangerous for both parties. If you can, avoid this and implement your API differently.

**Reply by sorasful** [reply]
Very clear, thank you for taking the time!
